src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiTypeAndSpec.cs

```diff
@@ -20,5 +20,36 @@ public class ApiTypeAndSpec
         [YamlMember(Alias = "specName")]
         [JsonProperty("specName")]
         public List<ApiLanguageValuePair> Spec { get; set; }
+
+        private bool _needExpand = true;
+
+        public static ApiTypeAndSpec FromUid(string uid)
+        {
+            if (string.IsNullOrEmpty(uid)) return null;
+
+            return new ApiTypeAndSpec
+            {
+                Uid = uid,
+            };
+        }
+
+        public static ApiTypeAndSpec FromUid(string uid, Dictionary<string, ApiReferenceBuildOutput> references, string[] supportedLanguages)
+        {
+            return new ApiTypeAndSpec
+            {
+                Uid = uid,
+                Spec = ApiBuildOutputUtility.GetSpec(uid, references, supportedLanguages),
+                _needExpand = false,
+            };
+        }
+
+        public void Expand(Dictionary<string, ApiReferenceBuildOutput> references, string[] supportedLanguages)
```

Comment:
Sounds good! add this later this sprint.

---
In reply to: [64332966](https://github.com/dotnet/docfx/pull/332#discussion_r64332966) [](ancestors = 64332966)